File: Persistence/Data/Contracts/IProductService.cs
using Entities;

namespace Contracts;

public interface IProductService
{
    public Task<IList<Product>> GetAllProductsAsync();
    public Task<Product> GetProductById(int id);
    public Task AddProductAsync(Product product);
    public Task DeleteProductByIdAsync(int id);
    public Task UpdateProductAsync(Product product);
}

File: Application/src/main/java/Contracts/IProductService.java
package Contracts;

import Entities.Product;

import java.util.ArrayList;

public interface IProductService {

    public ArrayList<Product> GetAllProducts();
    public Product GetProductById(int id);
    public void AddProduct(Product product); //TODO
    public void DeleteProductByIdAsync(int id);
    public void UpdateProductAsync(Product product);

}
